kentik_api_library/tests/integration/test_users.py

```diff
@@ -0,0 +1,302 @@
+import json
+from http import HTTPStatus
+
+import httpretty
+import pytest
+
+from kentik_api import KentikAPI
+from kentik_api.public.errors import (
+    AuthError,
+    BadRequestError,
+    NotFoundError,
+    RateLimitExceededError,
+    UnavailabilityError,
+    KentikAPIError,
+)
+from kentik_api.public.user import User
+
+AUTH_EMAIL_KEY: str = "X-CH-Auth-Email"
+AUTH_API_TOKEN_KEY: str = "X-CH-Auth-API-Token"
+DUMMY_AUTH_EMAIL: str = "email@example.com"
+DUMMY_TOKEN: str = "api-test-token"
+DUMMY_USER_ID: int = 1337
+FAKE_API_URL: str = "https://api.fakekentik.com/api/v5"
+
+
+def test_create_user_fails_when_no_email_given(kentik_api) -> None:
+    user = User(role="Member", email_service=True, email_product=True)
+
+    with pytest.raises(AssertionError, match="User email has to be provided"):
```

Comment:
once we are through with the incoming demo we should replace such assertions in code with IncompleteObject error and that should be checked in respective payload modules. Buy first throttling and demo

Reply:
Created ticket for that: https://clkentik.atlassian.net/browse/KNTK-152